### ISSUE TITLE
Can't set skipper maxTimeToBuffer and maxTimeToWaitForFirstFile options

### ISSUE THREAD

**Issue by Winangel** [balderdashy/skipper#131]
Hi,

I try to change  maxTimeToBuffer and maxTimeToWaitForFirstFile but it doesn't seem to work...

To do so I changed my config/http.js file:

``` javascript
module.exports.http ={
 bodyParser: function () {
    var opts = {
        maxTimeToBuffer:10000,
        maxTimeToWaitForFirstFile: 30000
      },
      fn;
    // Default to built-in bodyParser:
    fn = require('skipper');
    return fn(opts);
  },
  middleware: {...}
}
```

but I still get these kind of errors:

```
Error: ETIMEOUT: An Upstream (`file`) timed out waiting for file(s). No files were sent after waiting 10000ms.
```

What really bother me is not the error itself but the fact that it is still triggered after que 10000ms timeout when I changed this value to 30000ms ...

I had a look at #120. Is this a real bug ? if yes do you plan to merge the fix or am I getting things completely wrong ?

Thanks

EDIT : after diving into the code it seems that upstream is called without the options, making it impossible to change these options.

``` javascript
req.file = function (fieldName) {
      var noopUpstream = new Upstream({
        noop: true
        /** this is where you should add the right options**/
      });
      noopUpstream.fieldName = 'NOOP_' + fieldName;
      return noopUpstream;
    };
```

If it is ok i'll do a pull request to solve this issue.

**Comment by Winangel** [comment]
Please consider #132 to resolve this issue.

**Comment by mikermcneil** [comment]
@Winangel thanks for following up!  @sgress454 or I will take a look at this asap.

**Comment by sailsbot** [comment]
Thanks for posting, @Winangel.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sailsbot** [comment]
@Winangel,@mikermcneil,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by drinchev** [comment]
Is there any workaround this problem? I'm also unable to set `maxTimeToBuffer`?

**Comment by sgress454** [comment]
This is fixed and published as Skipper v0.7.3.  Thanks everybody!
